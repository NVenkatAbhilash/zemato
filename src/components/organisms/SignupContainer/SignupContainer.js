import React, { useState } from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import { Link } from "react-router-dom";
import Grid from "@material-ui/core/Grid";
import Typography from "../../atoms/Typography/Typography";
import { makeStyles } from "@material-ui/core/styles";
import auth from "../../../auth/initAuth";
import CloseIcon from "@material-ui/icons/Close";

const useStyles = makeStyles((theme) => ({
  container: {
    backgroundColor: "white",
    borderRadius: "10px",
    padding: "40px 20px",
    width: "500px",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(1),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
    borderRadius: "10px",
    opacity: ".8",
    textTransform: "capitalize",
    padding: "5px",
    fontSize: "22px",
  },
  google: {
    border: "1px solid rgb(100,100,100,.4)",
    padding: "18px",
    borderRadius: "10px",
    margin: "20px 0",
    textTransform: "capitalize",
    fontSize: "26px",
    opacity: ".7",
  },
  red: {
    color: "rgb(255,0,0,.7)",
    paddingLeft: "10px",
  },

  popup: {
    //display: "none",
    position: "fixed",
    zIndex: 1,
    left: 0,
    top: 0,
    paddingTop: "10vh",
    backgroundColor: "rgb(0,0,0,.9)",
    width: "100vw",
    height: "100vh",
  },
  link: {
    display: "inline",
    cursor: "pointer",
  },
}));

const SignupContainer = function ({setDisplay, setOtherDisplay ,...props}) {
  const classes = useStyles();

  const [email, setEmail] = useState("");

  const [password, setPassword] = useState("");

  const handleEmailChange = (e) => {
    setEmail(e.target.value);
    //console.log(email);
  };

  const handleLoginWithGoogle = (e) => {
    e.preventDefault();
    auth.loginWithGoogle();
  };

  const handlePasswordChange = (e) => {
    setPassword(e.target.value);
  };

  const handleSubmit = (e, data) => {
    e.preventDefault();
    //console.log(email+" "+password);
    auth.signup(email, password);
  };

  /*
  let display = false;
  //console.log();
  if(props.location.search.indexOf("login")!==-1){
    display = true;
  }

        style={{ display: display ? "block" : "none" }}

  */

  return (
    <div className={classes.popup} {...props}>
      <Grid container justify="center">
          <Grid container className={classes.container}>
            <Grid item container alignItems="baseline">
              <Grid item sm>
                <Typography variant="h4">Signup</Typography>
              </Grid>
              <Grid item>
                <CloseIcon
                  fontSize="small"
                  onClick={() => setDisplay(false)}
                  data-testid="close"
                />
              </Grid>
            </Grid>

            <form className={classes.form} noValidate>
              <TextField
                variant="outlined"
                margin="normal"
                required
                fullWidth
                label="Email Address"
                name="email"
                autoComplete="email"
                autoFocus
                onKeyUp={handleEmailChange}
                data-testid="email"
              />
              <TextField
                variant="outlined"
                margin="normal"
                required
                fullWidth
                name="password"
                label="Password"
                type="password"
                autoComplete="current-password"
                onKeyUp={handlePasswordChange}
                data-testid="password"
              />
              <Button
                type="submit"
                fullWidth
                variant="contained"
                color="primary"
                className={classes.submit}
                onClick={handleSubmit}
                data-testid="button"
              >
                Create Account
              </Button>
              <Grid container justify="center">
                <Grid item>
                  <Typography variant="h6">or</Typography>
                </Grid>
              </Grid>

              <Button
                type="submit"
                fullWidth
                className={classes.google}
                onClick={handleLoginWithGoogle}
                data-testid="googleLogin"
              >
                Continue with Google
              </Button>
              <Grid container>
                <Grid item>
                  <Typography>Already have an account? </Typography>
                 </Grid>
                 <Grid item>
                  <div
                    className={classes.link}
                    onClick={() => {
                      setDisplay(false);
                      setOtherDisplay(true);
                    }}
                    data-testid="login"
                  >
                    <Typography>
                      <span className={classes.red}> Login</span>
                    </Typography>
                  </div>
                </Grid>
              </Grid>
            </form>
          </Grid>
        </Grid>
    </div>
  );
};

export default SignupContainer;
