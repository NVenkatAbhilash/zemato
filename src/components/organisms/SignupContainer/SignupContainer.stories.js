import React from "react";
import SignupContainer from "./SignupContainer";

export default {
    title: "organisms/SignupContainer",
    component: SignupContainer,
}


const Template = ({...args})=>(
    <SignupContainer {...args} />
);


export const Primary = Template.bind({});
Primary.args = {
}
