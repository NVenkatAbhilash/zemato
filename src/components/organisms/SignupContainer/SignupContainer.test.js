import React from "react";
import ReactDOM from "react-dom";
import { render, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import SignupContainer from "./SignupContainer";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";


afterEach(cleanup);


it("renders SignupContainer snapshot correctly", () => {
  const tree = renderer
    .create(
        <SignupContainer />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders SignupContainer without crashing", () => {
  const div = document.createElement("div");
  const props = {
    setDisplay: () => true,
    setOtherDisplay: () => true,
  };
  ReactDOM.render(<SignupContainer props />, div);
});

it("renders SignupContainer correctly with email and password", () => {
  const props = {
    setDisplay: (something) => something,
    setOtherDisplay: (something) => something,
  };
  const { getByTestId } = render(<SignupContainer props />);
  const email = getByTestId("email").querySelector("input");
  //fireEvent.change(email, { target: { value: "abhilash@gmail.com" } });
  userEvent.type(email, "abhilash@gmail.com");
  expect(email.value).toEqual("abhilash@gmail.com");
  const password = getByTestId("password").querySelector("input");
  //  fireEvent.change(password, { target: { value: "Password123" } });
  userEvent.type(password, "Password123");
  expect(password.value).toEqual("Password123");
  fireEvent.click(getByTestId("button"));
});

it("renders SignupContainer continue with google button", () => {
  const props = {
    setDisplay: (something) => something,
    setOtherDisplay: (something) => something,
  };
  const { getByTestId } = render(<SignupContainer props />);
  expect(getByTestId("googleLogin")).toBeDefined();
});

it("renders SignupContainer Sign up button", () => {
  const props = {
    setDisplay: (something) => something,
    setOtherDisplay: (something) => something,
  };
  const { getByTestId } = render(<SignupContainer props />);
  expect(getByTestId("login")).toBeDefined();
});


const mockSetDisplay = jest.fn((x) => x);
const mockSetOtherDisplay = jest.fn((x) => x);

it("renders SignupContainer Sign up button", () => {
  const { getByTestId } = render(<SignupContainer setDisplay={mockSetDisplay} setOtherDisplay={mockSetOtherDisplay} />);
  expect(getByTestId("login")).toBeDefined();
  fireEvent.click(getByTestId("login"));
  expect(mockSetDisplay.mock.calls.length).toBe(1);
  expect(mockSetOtherDisplay.mock.calls.length).toBe(1);
});


it("renders SignupContainer close icon", () => {
  const { getByTestId } = render(<SignupContainer setDisplay={mockSetDisplay} setOtherDisplay={mockSetOtherDisplay} />);
  expect(getByTestId("close")).toBeDefined();
  fireEvent.click(getByTestId("close"));
  expect(mockSetDisplay.mock.calls.length).toBe(1);
});