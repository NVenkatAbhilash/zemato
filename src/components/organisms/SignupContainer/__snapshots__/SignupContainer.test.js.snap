// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignupContainer snapshot correctly 1`] = `
<div
  className="makeStyles-popup-6"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-container-1 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-baseline"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-true"
        >
          <h4
            className="MuiTypography-root MuiTypography-h4"
          >
            Signup
          </h4>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            data-testid="close"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
      </div>
      <form
        className="makeStyles-form-2"
        noValidate={true}
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          data-testid="email"
          onKeyUp={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
          >
            Email Address
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="email"
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Email Address
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          data-testid="password"
          onKeyUp={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
          >
            Password
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="current-password"
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Password
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-3 MuiButton-containedPrimary MuiButton-fullWidth"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Create Account
          </span>
        </button>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h6
              className="MuiTypography-root MuiTypography-h6"
            >
              or
            </h6>
          </div>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-google-4 MuiButton-fullWidth"
          data-testid="googleLogin"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Continue with Google
          </span>
        </button>
        <div
          className="MuiGrid-root MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Already have an account? 
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <div
              className="makeStyles-link-7"
              data-testid="login"
              onClick={[Function]}
            >
              <p
                className="MuiTypography-root MuiTypography-body1"
              >
                <span
                  className="makeStyles-red-5"
                >
                   Login
                </span>
              </p>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
